### PR TITLE
[janitor] Refactor listPop in list.c

### PR DESCRIPTION
**Pull request by phannebohm**
We popped from one list and pushed to another, each time freeing and allocating a node.
That was silly so I made it so the list reuses the popped node.

### CODE REVIEW

File: OMCompiler/SimulationRuntime/c/util/list.c [hunk 1]
```diff
@@ -150,20 +168,43 @@ void *listLastData(LIST *list)
   return list->last->data;
 }
 
-void listPopFront(LIST *list)
+/**
+ * @brief Return first node and pop from list
```
Comment (AnHeuermann) [inline OMCompiler/SimulationRuntime/c/util/list.c:172]: Make it clear, that memory for the returned node has to be freed with `freeNode` by the caller of `listPopFrontNode`.

**Review by AnHeuermann — CHANGES_REQUESTED** [review]
Looking good.

While you are on it, could you add a documentation to the rest of the list functions?
A clarification for `listPushFront` and `listPushFrontNodeNoCopy` would be great, so that is clear for everybody what they are doing/ what's the difference between them.

File: OMCompiler/SimulationRuntime/c/util/list.c (hunk 1 above)
Comment (phannebohm) [reply OMCompiler/SimulationRuntime/c/util/list.c:172]: I wrote it further down at the `@return` description but I guess nobody looks at that :smiling_face_with_tear:

Comment (AnHeuermann) [reply OMCompiler/SimulationRuntime/c/util/list.c:172]: Indeed I did not read it^^. Okay, that's my fault.

**Review by AnHeuermann — APPROVED** [review]

**Pull request merged**